misc/dbt-materialize/dbt/adapters/materialize/connections.py

```diff
@@ -65,24 +90,17 @@ def open(cls, connection):
         # More info: https://www.psycopg.org/docs/usage.html#transactions-control
         connection.handle.autocommit = True
 
-        cursor = connection.handle.cursor()
-
-        # Check for the current DB version using the "mz_introspection" cluster in case "default"
-        # doesn't exist.
-        cursor.execute("SHOW mz_version")
-        mz_version = cursor.fetchone()[0].split()[0].strip("v")
-
+        mz_version = connection.handle.info.parameter_status(
+            "mz_version"
+        )  # e.g. v0.79.0-dev (937dfde5e)
+        mz_version = mz_version.split()[0]  # e.g. v0.79.0-dev
+        mz_version = mz_version.removeprefix("v")  # e.g. 0.79.0-dev
```

Comment:
`removeprefix` requires Python >= 3.9, but we should strive for compatibility with Python 3.8 through 3.11 (same as [`dbt-postgres`](https://github.com/dbt-labs/dbt-core/blob/6e0a38720529360d061bd4e24e140f5ec55f8c15/plugins/postgres/setup.py#L77)). I'll uglify this bit!

Reply:
I wouldn't have thought of refactoring any of this — so thank you!